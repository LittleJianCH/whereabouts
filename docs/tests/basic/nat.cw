Nat "zero"

Nat { prev }
------------ {
  Nat prev
}

Zero "zero"

One { prev }
------------ {
  Zero prev
}

Two { prev }
------------ {
  One prev
}

Add ["zero", y, y]

Add [{ prev }, y, { prev: result_prev } ]
---------------------------------------- {
  Add [prev, y, result_prev]
}

query (z) {
  One x
  One y
  Add [x, y, z]
}

// TODO need limit

// query (x, y, z) {
//   Add [x, y, z]
// }

query (x, y) {
  Two z
  Add [x, y, z]
}

Mul ["zero", y, "zero"]

Mul [{ prev }, y, result]
---------------------------- {
  Add [z, y, result]
  Mul [prev, y, z]
}

query (z) {
  Two x
  Two y
  Mul [x, y, z]
}
