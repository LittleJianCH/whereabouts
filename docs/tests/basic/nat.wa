Nat "zero"

Nat { prev }
------------ {
  Nat prev
}

Zero "zero"

One { prev }
------------ {
  Zero prev
}

Two { prev }
------------ {
  One prev
}

Add ["zero", y, y]

Add [{ prev }, y, { prev: result_prev } ]
---------------------------------------- {
  Add [prev, y, result_prev]
}

query z {
  One x
  One y
  Add [x, y, z]
}

// TODO need limit

// query (x, y, z) {
//   Add [x, y, z]
// }

query (x, y) {
  Two z
  Add [x, y, z]
}

Mul ["zero", y, "zero"]

Mul [{ prev }, "zero", "zero"]

Mul [{ prev: prev_x }, { prev: prev_y } , result]
---------------------------- {
  Add [{ prev: prev_y }, z, result]
  Mul [prev_x, { prev: prev_y }, z]
}

query z {
  Two x
  Two y
  Mul [x, y, z]
}

query (x, y) {
  Two z
  Mul [x, y, z]
}
