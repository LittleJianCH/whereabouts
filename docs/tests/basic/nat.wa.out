{ "success": true, "count": 1, "solutions": [[{ "prev": { "prev": "zero" } }]] }
{ "success": true, "count": 3, "solutions": [["zero", { "prev": { "prev": "zero" } }], [{ "prev": "zero" }, { "prev": "zero" }], [{ "prev": { "prev": "zero" } }, "zero"]] }
{ "success": true, "count": 1, "solutions": [[{ "prev": { "prev": { "prev": { "prev": "zero" } } } }]] }
{ "success": true, "count": 2, "solutions": [[{ "prev": "zero" }, { "prev": { "prev": "zero" } }], [{ "prev": { "prev": "zero" } }, { "prev": "zero" }]] }
